### ISSUE TITLE
embedding issues

### ISSUE THREAD

**Issue by JeffBezanson** [JuliaLang/julia#21137]
Trying to run the embedding example I'm hitting a number of issues. Some I can fix myself (in an upcoming PR) but others I'll need help with:

- Running `make` gives
```
/home/jeff/src/julia/Make.inc:120: *** cowardly refusing to build from source directory with a shell-metacharacter in the path (got: $(abspath ..)).  Stop.
```
- The world age is 0, causing e.g. the `jl_show` call in embedding.c to fail. Adding `jl_get_ptls_states()->world_age = jl_world_counter;` to `jl_init` fixes it, but I wouldn't expect that to be necessary. Where should this initialization happen?

These I can fix unless there are objections:
- Need to add JCPPFLAGS and JCFLAGS to examples/Makefile.
- Default system image path when calling `jl_init` is broken.

**Comment by ararslan** [comment]
cc @tkelman and @staticfloat for point 1, @vtjnash for point 2

**Comment by vtjnash** [comment]
> the `jl_show` call

...should be gotten rid of

we don't need yet another entry point now that we have `jl_call` and `cfunction`

**Comment by staticfloat** [comment]
@JeffBezanson see https://github.com/JuliaLang/julia/pull/21140 for the incredibly in-depth and elaborate workaround required to fix part one.

**Comment by JeffBezanson** [comment]
Fixed by #21299
